DynamicTablesPkg/Library/Acpi/Common/AcpiFadtLib/FadtGenerator.c

```diff
@@ -551,6 +643,246 @@ BuildFadtTable (
+  Status = GetEArchCommonObjFadtMiscInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &FadtMiscInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get various platform information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    AcpiFadt.PLvl2Lat    = FadtMiscInfo->PLvl2Lat;
+    AcpiFadt.PLvl3Lat    = FadtMiscInfo->PLvl3Lat;
+    AcpiFadt.FlushSize   = FadtMiscInfo->FlushSize;
+    AcpiFadt.FlushStride = FadtMiscInfo->FlushStride;
+    AcpiFadt.DutyOffset  = FadtMiscInfo->DutyOffset;
+    AcpiFadt.DutyWidth   = FadtMiscInfo->DutyWidth;
+    AcpiFadt.DayAlrm     = FadtMiscInfo->DayAlrm;
+    AcpiFadt.MonAlrm     = FadtMiscInfo->MonAlrm;
+    AcpiFadt.Century     = FadtMiscInfo->Century;
+  }
+
+  // Get the 64-bit GPE Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtXgpeBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &XgpeBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get 64-bit GPE Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &AcpiFadt.XGpe0Blk,
+      &XgpeBlockInfo->XGpe0Blk,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    CopyMem (
+      &AcpiFadt.XGpe1Blk,
+      &XgpeBlockInfo->XGpe1Blk,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+  }
+
+  // Get the sleep Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtSleepBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &SleepBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get Sleep Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &AcpiFadt.SleepControlReg,
+      &SleepBlockInfo->SleepControlReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    CopyMem (
+      &AcpiFadt.SleepStatusReg,
+      &SleepBlockInfo->SleepStatusReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+  }
+
+  // Get the sleep Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtResetBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &ResetBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get Reset Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &AcpiFadt.ResetReg,
+      &ResetBlockInfo->ResetReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    AcpiFadt.ResetValue = ResetBlockInfo->ResetValue;
+  }
+

```

Comment:
It looks like in your case the FadtArchUpdate() from FadtGeneratorNull.c would be called. As I see it would return EFI_UNSUPPORTED which will result in table generation failure. Either this or it looks like some files are missing in your patch. Can you check, please?